rstudio/rstudio
# avoid auto-loading yaml, tinytex packages

### Intent

On startup, we were auto-loading the `yaml` and `tinytex` packages. This is undesirable, especially on Windows, as it makes it very challenging to update these packages within an RStudio session.

### Approach

- `tinytex`: use our own shim for finding the root of a `tinytex` installation, modeled after the function itself.
- `yaml`: use [yaml-cpp](https://github.com/jbeder/yaml-cpp) rather than the R `yaml` package for parsing YAML when enumerating tutorials.

### QA Notes

First, ensure the `tinytex` and `yaml` packages are installed:

```
install.packages(c("tinytex", "yaml"))
```

Then, open the Tutorial pane, and restart the R session. After restarting the R session, check the output of:

```
loadedNamespaces()
```

Confirm that neither `yaml` nor `tinytex` are in the list of installed packages.

---

Closes https://github.com/rstudio/rstudio/issues/8698.
